cmd/zfs/zfs_main.c

```diff
@@ -4325,6 +4330,23 @@ zfs_do_send(int argc, char **argv)
 		}
 	}
 
+	if (flags.partial) {
+		if (fromname != NULL || flags.replicate || flags.props ||
+		    flags.doall || flags.backup || flags.dedup ||
+		    flags.holds || redactbook != NULL) {
+			(void) fprintf(stderr,
+			    gettext("incompatible flags combined with partial "
+			    "send flag\n"));
+			usage(B_FALSE);
+		}
+		if (strchr(argv[0], '@') != NULL) {
+			(void) fprintf(stderr,
+			    gettext("partial send cannot send full "
+			    "snapshots\n"));
+			usage(B_FALSE);
```

Comment:
Let's add these invalid parameter settings to the existing `tests/functional/cli_root/zfs_send/zfs_send_004_neg.ksh` test case.

Reply:
Would it make more sense to add these to the partial_send test? Every combination of commands will fail if you don't have a partially received dataset, which `zfs_send_004_neg` doesn't have at the moment.

Reply:
That's a good point.  I think either location would be fine, if you'd prefer to add it to the partial_send test since it's more convenient that's fine by me.